### PR TITLE
Fix _cluster_setup api | couchdb node name changes | verify cluster setup  

### PR DESCRIPTION
**Pull request by raghavtan**
- When doing cluster_setup via api or wizard we need to make changes in vm.args as all nodes should have a unique name (mentioned in the file comments but not in the cluster setup documentation)
- executing _cluster_setup api with enable_cluster the second time gives error response on the coordination node when adding nodes to cluster hence is not required causes failure
- only add_node action is required from _cluster_setup api on  coordination node
- Added cluster setup verification docs

### CODE REVIEW

File: src/cluster/setup.rst [hunk 1]
```diff
@@ -167,6 +167,10 @@ Before you can add nodes to form a cluster, you have to have them
 listen on a public IP address and set up an admin user. Do this, once
 per node:
 
+Change line ``-name couchdb@127.0.0.1`` in file /Couch-Home/etc/vm.args on
```
Comment (wohali) [inline src/cluster/setup.rst]: Just specify the path here as `etc/vm.args`, please.

File: src/cluster/setup.rst [hunk 2]
```diff
@@ -167,6 +167,10 @@ Before you can add nodes to form a cluster, you have to have them
...
+each node for clustered setup, each node in system must have a unique name.
+eg. ``-name couchdb@couch1`` | ``-name couchdb@couch2``
```
Comment (wohali) [inline src/cluster/setup.rst]: The `couch1`, `couch2` hostnames must be fully qualified domain names (FQDNs) with valid DNS entries, see [the comment in vm.args](https://github.com/apache/couchdb/blob/master/rel/overlay/etc/vm.args#L13-L28) for more detail.

File: src/cluster/setup.rst [hunk 3]
```diff
@@ -211,12 +215,11 @@ requires all other nodes to be able to see it and vice versa.
 Set up will not work with unavailable nodes.
 The notion of "setup coordination node" will be gone once the setup is finished.
 From then on, the cluster will no longer have a "setup coordination node".
-To add a node run these two commands:
+To add a node run this commands for each node you want to add:
 
 .. code-block:: bash
 
-    curl -X POST -H "Content-Type: application/json" http://admin:password@127.0.0.1:5984/_cluster_setup -d '{"action": "enable_cluster", "bind_address":"0.0.0.0", "username": "admin", "password":"password", "port": 15984, "node_count": "3", "remote_node": "<remote-node-ip>", "remote_current_user": "<remote-node-username>", "remote_current_password": "<remote-node-password>" }'
```
Comment (wohali) [inline src/cluster/setup.rst]: No, actually, you need this first statement too. This sets the bind address, bind port, admin username and admin password on the remote node. The node_count parameter is also critical.

You can see a test cluster being set up in the test case for the `couch_setup` application [here](https://github.com/apache/couchdb-setup/blob/master/test/t-frontend-setup.sh) and [here](https://github.com/apache/couchdb-setup/blob/master/test/t.sh).

File: src/cluster/setup.rst [hunk 4]
```diff
@@ -211,12 +215,11 @@ requires all other nodes to be able to see it and vice versa.
...
-    curl -X POST -H "Content-Type: application/json" http://admin:password@127.0.0.1:5984/_cluster_setup -d '{"action": "add_node", "host":"<remote-node-ip>", "port": "<remote-node-port>", "username": "admin", "password":"password"}'
+    curl -X POST -H "Content-Type: application/json" http://admin:password@127.0.0.1:5984/_cluster_setup -d '{"action": "add_node", "host":"<remote-node-ip>", "port": <remote-node-port>, "username": "admin", "password":"password"}'
```
Comment (wohali) [inline src/cluster/setup.rst]: You're also missing a quote mark here. The original line was correct.

File: src/cluster/setup.rst [hunk 5]
```diff
@@ -227,6 +230,39 @@ following command to complete the setup and add the missing databases:
...
+        "all_nodes": [
+            "couchdb@couch1",
+            "couchdb@couch2",
+        ],
+        "cluster_nodes": [
+            "couchdb@couch1",
+            "couchdb@couch2",
+        ]
+    }
+
```
Comment (wohali) [inline src/cluster/setup.rst]: This is a useful thing to add, thanks.

File: src/cluster/setup.rst [hunk 6]
```diff
@@ -167,6 +167,9 @@ Before you can add nodes to form a cluster, you have to have them
 listen on a public IP address and set up an admin user. Do this, once
 per node:
 
+Change line ``-name couchdb@127.0.0.1`` in file etc/vm.args on
+each node for clustered setup, each node in system must have a unique name.
```
Comment (flimzy) [inline src/cluster/setup.rst]: This is a run-on sentence. I suggest changing to:

> ... for clustered setup. Each node in the system must have a unique name.

File: src/cluster/setup.rst [hunk 7]
```diff
@@ -211,12 +214,11 @@ requires all other nodes to be able to see it and vice versa.
...
 The notion of "setup coordination node" will be gone once the setup is finished.
 From then on, the cluster will no longer have a "setup coordination node".
-To add a node run these two commands:
+To add a node run these commands for each node you want to add:
 
 .. code-block:: bash
-
     curl -X POST -H "Content-Type: application/json" http://admin:password@127.0.0.1:5984/_cluster_setup -d '{"action": "enable_cluster", "bind_address":"0.0.0.0", "username": "admin", "password":"password", "port": 15984, "node_count": "3", "remote_node": "<remote-node-ip>", "remote_current_user": "<remote-node-username>", "remote_current_password": "<remote-node-password>" }'
-    curl -X POST -H "Content-Type: application/json" http://admin:password@127.0.0.1:5984/_cluster_setup -d '{"action": "add_node", "host":"<remote-node-ip>", "port": "<remote-node-port>", "username": "admin", "password":"password"}'
+    curl -X POST -H "Content-Type: application/json" http://admin:password@127.0.0.1:5984/_cluster_setup -d '{"action": "add_node", "host":"<remote-node-ip>", "port": <remote-node-port>, "username": "admin", "password":"password"}'
```
Comment (flimzy) [inline src/cluster/setup.rst]: Are you sure the port doesn't need to be a string? AFAIK, all config values are strings. Not to mention, a port can (probably) be non-numeric, i.e. `http` or `https`, etc.

Comment (raghavtan) [reply src/cluster/setup.rst]: @flimzy sure of that the port doesnt need to be string.

File: src/cluster/setup.rst (hunk 5 above)
Comment (raghavtan) [reply src/cluster/setup.rst]: np

File: src/cluster/setup.rst [hunk 8]
```diff
@@ -167,6 +167,9 @@ Before you can add nodes to form a cluster, you have to have them
 listen on a public IP address and set up an admin user. Do this, once
 per node:
 
+``-name couchdb@127.0.0.1`` in file etc/vm.args on
```
Comment (flimzy) [inline src/cluster/setup.rst]: Now this is an incomplete sentence :)

File: src/cluster/setup.rst [hunk 9]
```diff
@@ -164,9 +164,15 @@ procedure on other machines. Be sure to specify the total number of nodes you
 expect to add to the cluster before adding nodes.
 
 Before you can add nodes to form a cluster, you have to have them
-listen on a public IP address and set up an admin user. Do this, once
+listen on a node address if all nodes can communicate over those IP's,
+also set up an admin user which should be same on all nodes. Do this, once
 per node:
```
Comment (flimzy) [inline src/cluster/setup.rst]: This is some very awkward wording, IMO. I suggest rewriting the paragraph as:

> Before you can add nodes to form a cluster, you must have them listening on an IP address accessible from the other nodes in the cluster.  Do this once per node:

> In file ``etc/vm.args`` change the the line ``-name couchdb@127.0.0.1`` to ``-name couchdb@<this-nodes-ip-address>`` for each node. For clustered setup, each node in the system must have a unique name.

Comment (raghavtan) [reply src/cluster/setup.rst]: Rephrased but added parameter <FQDN> along with IP @flimzy

File: src/cluster/setup.rst [hunk 10]
```diff
@@ -163,9 +163,17 @@ then to add nodes by IP address. To get more nodes, go through the same install
...
 
-Before you can add nodes to form a cluster, you have to have them
-listen on a public IP address and set up an admin user. Do this, once
-per node:
+Before you can add nodes to form a cluster, you must have them listening on an
+IP address accessible from the other nodes in the cluster.
+Do this once per node:
+
+In file etc/vm.args  to -name couchdb@<this-nodes-ip-address> for each node.
+For clustered setup, each node in the system must have a unique name.
```
Comment (flimzy) [inline src/cluster/setup.rst]: This paragraph looks like a partial copy of the following one.  Copy and paste error?

Comment (wohali) [reply src/cluster/setup.rst]: @raghavtan can you check this comment from @flimzy before I squash and merge? thanks.

**Comment by raghavtan** [comment]
@wohali fixed

**Comment by wohali** [comment]
@raghavtan Thanks again, and congratulations on your first contribution to Apache CouchDB!

**Pull request merged**